var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -44,15 +45,35 @@ class Boost(Package):
     version('1.34.1', '2d938467e8a448a2c9763e0a9f8ca7e5')
     version('1.34.0', 'ed5b9291ffad776f8757a916e1726ad0')
 
+    libs = ['chrono', 
+        'date_time', 
+        'filesystem', 
+        'iostreams', 
+        'random', 
+        'regex', 
+        'serialization', 
+        'signals', 
+        'system', 
+        'thread', 
+        'wave', 
+        'mpi', 
```

Comment:
Same thing here:

```
spack install boost +mpi ^openmpi
spack install boost +mpi ^mpich
spack install boost +mpi ^mvapich2
```

etc.  You know the version you're building with from the `spec` so you can figure out how best to set the args.